frame/treasury/src/lib.rs

```diff
@@ -153,6 +184,25 @@ pub mod pallet {
 		/// Origin from which rejections must come.
 		type RejectOrigin: EnsureOrigin<Self::RuntimeOrigin>;
 
+		type Balance: Balance;
+
+		/// The identifier for what asset should be spent.
+		type AssetKind: AssetId;
+
+		/// Means by which we can make payments to accounts. This also defines the currency and the
+		/// balance which we use to denote that currency.
+		type Paymaster: Pay<
+			Beneficiary = <Self as frame_system::Config>::AccountId,
+			AssetKind = Self::AssetKind,
+		>;
+
+		type BalanceConverter: BalanceConversion<
+			PayBalanceOf<Self, I>,
+			Self::AssetKind,
+			Self::Balance,
+			Error = Error<Self, I>,
```

Comment:
We should not be required to define `Error` here. That's the reason why it's used as an associated type of `BalanceConverter` instead of a generic.

Right now, you definitely require to have `pallet::Error::<T>::UnknownConversion` and map any error of `BalanceConverter` to that. By not specifying the `BalanceConverter<Error>`, you can remove `UnknownConversion` and actually bubble up.

In the end, it depends on whether you want to re-route the error or not.

Reply:
Yeah, I've removed it. Thanks